### ISSUE TITLE
An assembly must have a 'dll' extension in order to be used as an analyzer

### ISSUE THREAD

**Issue by provegard** [dotnet/roslyn#931]
I installed Visual Studio 2015 Preview, then ran `Install-Package Microsoft.CodeAnalysis -Pre` in the Package Manager console. While the output says "Successfully installed 'Microsoft.CodeAnalysis 1.0.0-rc1'", I also got the following:

```
Exception calling "Add" with "1" argument(s): "An assembly must have a 'dll' extension in order to be used as an analyzer."
At C:\users\per\documents\visual studio 2015\Projects\roslyn-lab\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ComMethodTargetInvocation
 
Exception calling "Add" with "1" argument(s): "An assembly must have a 'dll' extension in order to be used as an analyzer."
At C:\users\per\documents\visual studio 2015\Projects\roslyn-lab\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ComMethodTargetInvocation
 ```

**Comment by provegard** [comment]
With Visual Studio 2013, I get these error instead:

```
You cannot call a method on a null-valued expression.
At C:\users\per\documents\visual studio 2013\Projects\roslyn-lab\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
You cannot call a method on a null-valued expression.
At C:\users\per\documents\visual studio 2013\Projects\roslyn-lab\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
You cannot call a method on a null-valued expression.
At C:\users\per\documents\visual studio 2013\Projects\roslyn-lab\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
You cannot call a method on a null-valued expression.
At C:\users\per\documents\visual studio 2013\Projects\roslyn-lab\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:17 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
```

**Comment by jaredpar** [comment]
Thanks for reporting this issue!

The errors in the second portion is a known issue tracked by #639.  I'm not familiar with the first set of errors though and will leave this open in case it represents a different issue we need to fix.

**Comment by tmeschter** [comment]
@jaredpar I don't think you mean issue #639.

@jmarolf Is this fallout from us simply attempting to add everything in the directory as an analyzer?

**Comment by jaredpar** [comment]
@tmeschter correct.  I meant #629.  

Another reminder that I shouldn't communicate with people until I've finished my first cup of coffee :frowning:

**Comment by srivatsn** [comment]
@jmarolf is aware of this and is fixing it already. While enumerating the files a -File is missing in the script and so it's trying to add the directory as an analyzer. When the script was written this error wasn't being thrown and hence wasn't caught.

**Comment by jmarolf** [comment]
@jaredpar this was fixed with an internal changeset 1424664. Could you please verify?

**Comment by lgolding** [comment]
Verified as fixed.
